hedera-node/hedera-mono-service/src/main/java/com/hedera/services/txns/crypto/CryptoCreateTransitionLogic.java

```diff
@@ -169,19 +228,87 @@ public ResponseCodeEnum validate(TransactionBody cryptoCreateTxn) {
         if (memoValidity != OK) {
             return memoValidity;
         }
-        if (!op.hasKey()) {
-            return KEY_REQUIRED;
+        boolean emptyAlias = op.getAlias().isEmpty();
+        if (!emptyAlias && op.getAlias().size() < EVM_ADDRESS_SIZE) {
+            return INVALID_ALIAS_KEY;
         }
-        if (!validator.hasGoodEncoding(op.getKey())) {
-            return BAD_ENCODING;
+
+        if (emptyAlias || op.hasKey()) {
+            if (!op.hasKey()) {
+                return KEY_REQUIRED;
+            }
+
+            final var keyValidity = validateKey(op);
+            if (keyValidity != OK) {
+                return keyValidity;
+            }
+
+            if (!op.getKey().getECDSASecp256K1().isEmpty()) {
+                var evmAddressValidity =
+                        tryToRecoverEVMAddressAndCheckValidity(
+                                op.getKey().getECDSASecp256K1().toByteArray());
+
+                if (evmAddressValidity != OK) {
+                    return evmAddressValidity;
+                }
+            }
```

Comment:
Lines 246-256 could be moved to a private method - `validateEVMAddressRecoverFromECDSAKey()` and it could be also used to substitute lines 269-277.